src/lightning/fabric/plugins/precision/amp.py

```diff
@@ -41,6 +41,9 @@ def __init__(
         device: str,
         scaler: Optional[torch.cuda.amp.GradScaler] = None,
     ) -> None:
+        if precision not in ("16-mixed", "bf16-mixed"):
+            raise ValueError(f"`{type(self).__name__}(precision={precision!r})` must be '16-mixed' or 'bf16-mixed'")
```

Comment:
Nit: Here you are saying "x(y) must be z" but what you really mean is "y must be z" (the parameter not the class the parameter is passed to).

Reply:
I suggested this. In my opinion, having the class that triggered the error in message is helpful to understand where this is coming from. Maybe this would clarify it

```suggestion
        if precision not in ("16-mixed", "bf16-mixed"):
            raise ValueError(f"Passed `{type(self).__name__}(precision={precision!r})`. Precision must be '16-mixed' or 'bf16-mixed'")
```

Reply:
Done